cli/src/pcluster/resources/imagebuilder/custom_script.yaml

```diff
@@ -0,0 +1,75 @@
+name: ScriptComponent
+description: ScriptComponent
+schemaVersion: 1.0
+
+phases:
+  - name: build
+    steps:
+      # Exit code for failures
+      - name: Fail
+        action: ExecuteBash
+        inputs:
+          commands:
+            - echo 1
+
+      # Script url
+      - name: ScriptUrl
+        action: ExecuteBash
+        inputs:
+          commands:
+            - |
+              set -v
+              echo {{script-url}}
+
+      # Script url scheme
+      - name: ScriptUrlScheme
+        action: ExecuteBash
+        inputs:
+          commands:
+            - |
+              set -v
+              echo {{script-url-scheme}}
+
+      # Create a temporary file
+      - name: TempScript
+        action: ExecuteBash
+        inputs:
+          commands:
+            - |
+              set -v
+              temp_file=$(mktemp --tmpdir parallelcluster.XXXXXXXXXX)
+              echo "${temp_file}"
+
+      # Download script
+      - name: Download
+        action: ExecuteBash
+        inputs:
+          commands:
+            - |
+              set -v
+              if [[ {{ build.ScriptUrlScheme.outputs.stdout }} == "https" ]]; then
+                curl --retry 3 -L -o {{ build.TempScript.outputs.stdout }} {{ build.ScriptUrl.outputs.stdout }}
+              elif [[ {{ build.ScriptUrlScheme.outputs.stdout }} == "s3" ]]; then
+                aws s3 cp {{ build.ScriptUrl.outputs.stdout }} {{ build.TempScript.outputs.stdout }}
+              else
+                echo "Invalid script url"
+                exit {{ build.Fail.outputs.stdout }}
+              fi
+
+      # Make script executable
+      - name: Executable
+        action: ExecuteBash
+        inputs:
+          commands:
+            - |
+              set -v
+              chmod +x {{ build.TempScript.outputs.stdout }} || (echo 'The bash script is not executable.' && exit {{ build.Fail.outputs.stdout }})
```

Comment:
`echo 'Unable to set script executable'`

Reply:
done